test/test_cuda_sync_guard.py

```diff
@@ -0,0 +1,42 @@
+# Owner(s): ["module: cuda"]
```

Comment:
Please move these checks to test_autograd no need for a special file

Reply:
You can move that to test/autograd/* folder then and make sure to set the appropriate flag in run_test.py to avoid parallelist.

@ngimel I guess it is expected for the CudaSyncGuard to be global and not thread local?